### DISCUSSION TITLE
Provide commands / bindable shortcuts to toggle more settings (particularly: various hints)

### DISCUSSION THREAD

**Discussion by ferdnyc** [microsoft/pylance-release#5876] [Enhancement]
Pylance has many settings to control the level of hinting that's performed when viewing Python code, but it appears that all of those are exclusively that: Settings.

The more of those settings I enable, the more often I find myself wishing I could quickly toggle one of them off.

Inlay hints are extremely helpful in _some_ codebases, but in others they become an almost incomprehensible amount of visual clutter. I'd love to be able to bind a keyboard shortcut to toggle them off, in those situations, or even just go to the Command Palette and run a "Pylance hide inlay hints" command, but it doesn't appear that's possible.

I notice that clangd has a number of commands which can also be bound to shortcuts, including `clangd.inlayHints.toggle`/"clangd: Toggle inlay hints". It'd be great if Pylance offered similar.

**Comment by rchiodo** [comment]
Key bindings would work if we had a command for 'Toggle Inlay Hints'. Then you could bind a key to it.

Turns out there's an extension that does exactly what you want written by @rebornix:
https://marketplace.visualstudio.com/items?itemName=rebornix.toggle

I believe that would accomplish what you want to do.

**Reply by ferdnyc** [reply]
Awesome, thanks!

Though, I have to wonder, is that a reason _not_ to provide the functionality directly within Pylance? The existence of an extension doesn't _preclude_ the feature being added, I'd think?

(I'm just wondering why the close. I was under the impression Discussions, unlike Issues, were the place to gauge community interest in possible new developments for the extension. Don't get me wrong, I'm not complaining or objecting, just curious. In terms of my own immediate need, that extension sounds like it'll serve just fine.)

**Reply by rchiodo** [reply]
I believe this can still be upvoted even it's closed. We close things to indicate we're not planning on implementing the feature.

**Reply by ferdnyc** [reply]
@rchiodo That makes sense, thanks for explaining!
